specs/agents/tracing-api-otel.md

```diff
@@ -0,0 +1,130 @@
+## OpenTelemetry API (Tracing)
+
+[OpenTelemetry](https://opentelemetry.io) (OTel in short) provides a vendor-neutral API that allows to capture tracing, logs and metrics data.
+
+Agents may provide a bridge implementation of OpenTelemetry Tracing API following this specification.
+
+Bridging here means that for each OTel span created with the API, a native span/transaction will be created and sent to APM server.
+
+From the perspective of the application code calling the OTel API, the delegation to a native span/transaction is transparent.
+Also, this means that any OTel processors will be bypassed and ignored by the bridge.
+
+### Spans and Transactions
+
+OTel only defines Spans, whereas Elastic APM relies on both Spans and Transactions.
+OTel allows users to provide the _remote context_ when creating a span, which is equivalent to providing a parent to a transaction or span,
+it also allows to provide a (local) parent span.
+
+As a result, when creating Spans through OTel API with a bridge, agents must implement the following algorithm:
+
+```javascript
+// otel_span contains the properties set through the OTel API
+span_or_transaction = null;
+if (otel_span.remote_contex != null) {
+    span_or_transaction = createTransactionWithParent(otel_span.remote_context);
```

Comment:
That's a very good point here, thanks for pointing that !

While we can't really know if the parent will be sent to ES and thus visible in Kibana, we could guess when it is coming from an Elastic agent, for example by parsing the `tracestate` header that should contain our vendor-specific key, but that would not work as expected with `service with Elastic > service with OTel > service with Elastic` where the `OTel` propagates upstream `tracestate` from the parent `Elastic` agent.

In other words, we need to have a way to know if the direct parent is properly captured by an Elastic agent or not.

Reply:
In the [W3c spec](https://www.w3.org/TR/trace-context/#relationship-between-the-headers), this is probably why the ID of the parent in `traceparent` is duplicated in `tracestate`. If we were doing that, it would be easy to detect when the parent is also captured with an Elastic agent or not.

Reply:
Interesting discussion but this is orthogonal to OTel. I think we can probably solve that in the context of https://github.com/elastic/apm/issues/286.

> OTel span which we may not have in elasticsearch

As the APM Server has an OTLP intake, the OTel spans may very well be in ES.